dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/globalid/DOIDataCiteRegisterService.java

```diff
@@ -87,18 +87,18 @@ public String reserveIdentifier(String identifier, Map<String, String> metadata,
 
     public void registerIdentifier(String identifier, Map<String, String> metadata, DvObject dvObject) throws IOException {
         String xmlMetadata = getMetadataFromDvObject(identifier, metadata, dvObject);
-        DOIDataCiteRegisterCache rc = findByDOI(identifier);
+        DOIDataCiteRegisterCache rc = Optional.ofNullable(findByDOI(identifier))
+                .orElseGet(DOIDataCiteRegisterCache::new);
```

Comment:
Files get registered directly on publication (whereas dataset we reserve the DOI when creating it). A cache entry was only been created by the reservation method. As a result DOI for files would never get unregistered because of the missing entry in the cache table.

Reply:
I'm not sure if this `DOIDataCiteRegisterCache` is even needed (for both datasets and files). Maybe I am missing something in the code.

Reply:
Yes we can add a follow-up issue for that.

Reply:
Issue created: https://github.com/CeON/dataverse/issues/2553